lib/decoder.rb

```diff
@@ -1,3 +1,11 @@
-class Decoder
+require './lib/crypto_tool'
 
+class Decoder < CryptoTool
+
+  def decrypt(ciphertext, key, date = nil)
+    date = today if date.nil?
+    shifts = shifts(keys(key), offset(date)).map{|shift| -shift}
```

Comment:
change each shift in `shifts` to its opposite (negative) causes the `caesar shift` to rotate "backwards", thus deciphering the character